### PR TITLE
[Spark-14300][Docs][MLLIB]Scala MLlib examples code merge and clean up

### PR DESCRIPTION
**Pull request by keypointt**
## What changes were proposed in this pull request?

https://issues.apache.org/jira/browse/SPARK-14300

Duplicated code found in scala/examples/mllib, below all deleted in this PR:
- DenseGaussianMixture.scala
- StreamingLinearRegression.scala
#### delete: mllib/DenseGaussianMixture.scala
- duplicate of mllib/GaussianMixtureExample
#### delete: mllib/StreamingLinearRegression.scala
- duplicate of mllib/StreamingLinearRegressionExample

When merging and cleaning those code, be sure not disturb the previous example on and off blocks.
## How was this patch tested?

Test with `SKIP_API=1 jekyll` manually to make sure that works well.

### CODE REVIEW

**Comment by keypointt** [comment]
#### Unsure code duplications (need doube check)

- AbstractParams.scala - cannot delete this one, not duplicate, its usage found in many examples

- BinaryClassification.scala - @pending: not sure, duplicate of mllib/BinaryClassificationMetricsExample ???
    
- CosineSimilarity.scala - @pending: personally I prefer to keep it as a demo
    
- MovieLensALS.scala - @pending: duplicate of LocalALS.scala, but I prefer to delete LocalALS.scala but keep MovieLensALS.scala, since MovieLensALS better commented
   
- MultivariateSummarizer.scala - @pending: usage not found, but better keep it for possible later demo
    
- RandomRDDGeneration.scala - @pending: usage not found, but better keep it for possible later demo
    
- SampledRDDs.scala - @pending: usage not found, but better keep it for possible later demo

hi @yinxusen, for above @pending ones, I'm not sure we should delete them or not? There are no usage found for them, but personally I find they are still useful examples

**Comment by yinxusen** [comment]
@keypointt As @mengxr said here https://github.com/apache/spark/pull/12092#issuecomment-204276885, we should keep examples such as `DecisionTreeRunner`.

**Comment by yinxusen** [comment]
So I think we should keep

* mllib/DecisionTreeRunner.scala
* mllib/GradientBoostedTreesRunner.scala
* mllib/DenseKMeans.scala
* mllib/LDAExample.scala
* mllib/LinearRegression.scala
* mllib/SparseNaiveBayes.scala

**Comment by yinxusen** [comment]
And for examples in **Unsure code duplications (need doube check)** you mentioned above, I think we should keep them, too.

**Comment by keypointt** [comment]
@yinxusen I've added back the files you mention above, and also updated the description (top comment) in this PR

**Comment by keypointt** [comment]
@yinxusen I've added back the files you mention above, and also updated the description (top comment) in this PR. In this PR I only deleted couple files, is there anything extra I should do?

**Comment by yinxusen** [comment]
@keypointt Sorry for the late. I'm  taking a look now.

**Comment by yinxusen** [comment]
@mengxr LGTM

**Comment by keypointt** [comment]
@yinxusen No problem, I totally understand you guys are super busy 😄

**Comment by keypointt** [comment]
ping @mengxr

File: examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLinearRegression.scala [hunk 1]
```diff
@@ -1,73 +0,0 @@
...
-// scalastyle:off println
-package org.apache.spark.examples.mllib
-
-import org.apache.spark.SparkConf
-import org.apache.spark.mllib.linalg.Vectors
-import org.apache.spark.mllib.regression.{LabeledPoint, StreamingLinearRegressionWithSGD}
-import org.apache.spark.streaming.{Seconds, StreamingContext}
-
-/**
- * Train a linear regression model on one stream of data and make predictions
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLinearRegression.scala]: It looks like this documentation was not copied to the replacement ```StreamingLinearRegressionExample```.  Could you please copy it there?

File: examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLogisticRegression.scala [hunk 2]
```diff
@@ -1,75 +0,0 @@
...
- *
- * To run on your local machine using the two directories `trainingDir` and `testDir`,
- * with updates every 5 seconds, and 2 features per data point, call:
- *    $ bin/run-example mllib.StreamingLogisticRegression trainingDir testDir 5 2
- *
- * As you add text files to `trainingDir` the model will continuously update.
- * Anytime you add text files to `testDir`, you'll see predictions from the current model.
- *
- */
-object StreamingLogisticRegression {
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLogisticRegression.scala]: I don't see this example anywhere else.  Am I missing something?  If not, then let's leave it.

File: examples/src/main/scala/org/apache/spark/examples/mllib/TallSkinnyPCA.scala [hunk 3]
```diff
@@ -1,62 +0,0 @@
...
- * The input matrix must be stored in row-oriented dense format, one line per row with its entries
- * separated by space. For example,
- * {{{
- * 0.5 1.0
- * 2.0 3.0
- * 4.0 5.0
- * }}}
- * represents a 3-by-2 matrix, whose first row is (0.5, 1.0).
- */
-object TallSkinnyPCA {
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/TallSkinnyPCA.scala]: Same here.  What is this duplicating?

File: examples/src/main/scala/org/apache/spark/examples/mllib/TallSkinnySVD.scala [hunk 4]
```diff
@@ -1,62 +0,0 @@
...
- * The input matrix must be stored in row-oriented dense format, one line per row with its entries
- * separated by space. For example,
- * {{{
- * 0.5 1.0
- * 2.0 3.0
- * 4.0 5.0
- * }}}
- * represents a 3-by-2 matrix, whose first row is (0.5, 1.0).
- */
-object TallSkinnySVD {
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/TallSkinnySVD.scala]: Ditto

**Review by jkbradley — COMMENTED** [review]
Sorry to leave your PR hanging, but if you have time now, it'd be good to fix up and merge.  Thanks!

File: examples/src/main/scala/org/apache/spark/examples/mllib/TallSkinnySVD.scala (hunk 4 above)
Comment (keypointt) [inline examples/src/main/scala/org/apache/spark/examples/mllib/TallSkinnySVD.scala]: sure Joseph, I'm adding these files back

**Comment by keypointt** [comment]
The main reason deleting those without duplication is because of the name, but the actual code differs:

* StreamingLinearRegression
https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLinearRegression.scala

* StreamingLinearRegressionExample
https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLinearRegressionExample.scala

**Comment by keypointt** [comment]
I'll update the PR description if it looks good now

**Comment by jkbradley** [comment]
@keypointt I think it's OK to delete StreamingLinearRegression.scala, as you were doing before your last commit.  For that one, my only complaint was that the similar example (StreamingLinearRegressionExample.scala) did not include the Scala doc describing the input data.  If you copy that doc into StreamingLinearRegressionExample.scala, then it will be OK to remove StreamingLinearRegression.scala.

**Comment by keypointt** [comment]
I've also updated the description of this PR

**Comment by jkbradley** [comment]
LGTM
merging with master
Thanks @keypointt !

**Comment by keypointt** [comment]
thank you @jkbradley for the detailed explanation :)

**Pull request closed without merging**